File: backend/app/schemas/resume_builder.py
from typing import List, Optional

from pydantic import BaseModel, Field


class ResumeExperience(BaseModel):
    company: str
    role: str
    duration: Optional[str] = None
    bullets: List[str] = Field(default_factory=list)


class ResumeProject(BaseModel):
    name: str
    description: str
    technologies: List[str] = Field(default_factory=list)


class ResumeEducation(BaseModel):
    institution: str
    degree: str
    field: Optional[str] = None
    duration: Optional[str] = None


class TailoredResume(BaseModel):
    name: str
    professional_summary: str

    skills: List[str] = Field(default_factory=list)

    experience: List[ResumeExperience] = Field(
        default_factory=list
    )

    projects: List[ResumeProject] = Field(
        default_factory=list
    )

    education: List[ResumeEducation] = Field(
        default_factory=list
    )

    certifications: List[str] = Field(
        default_factory=list
    )


class ResumeChange(BaseModel):
    section: str
    original: str
    optimized: str
    reason: str


class ResumeBuilderResult(BaseModel):
    resume: TailoredResume

    ats_score: Optional[int] = None

    changes: List[ResumeChange] = Field(
        default_factory=list
    )

File: frontend/src/types/resumeBuilder.ts
export interface ResumeExperience {
  company: string;
  role: string;
  duration?: string;
  bullets: string[];
}

export interface ResumeProject {
  name: string;
  description: string;
  technologies: string[];
}

export interface ResumeEducation {
  institution: string;
  degree: string;
  field?: string;
  duration?: string;
}

export interface TailoredResume {
  name: string;
  professional_summary: string;
  skills: string[];
  experience: ResumeExperience[];
  projects: ResumeProject[];
  education: ResumeEducation[];
  certifications: string[];
}

export interface ResumeChange {
  section: string;
  original: string;
  optimized: string;
  reason: string;
}

export interface ResumeBuilderResult {
  resume: TailoredResume;
  ats_score: number | null;
  changes: ResumeChange[];
}

export interface ResumeBuilderResponse {
  status: string;
  resume_id: number;
  job_id: number;
  result: ResumeBuilderResult;
}

export interface SavedTailoredResume {
  resume_id: number;
  job_id: number;
  resume_data: TailoredResume;
  target_job_title?: string;
  target_company?: string;
  title?: string;
  saved_at: string;
}

export interface SaveResumeRequest {
  resume_id: number;
  job_id: number;
  resume_data: TailoredResume;
  target_job_title?: string;
  target_company?: string;
  title?: string;
}

export interface SaveResumeResponse {
  status: string;
  message: string;
  saved_resume: SavedTailoredResume;
}
